plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -198,8 +198,55 @@ public function __construct( $args = array() ) {
 		// TODO: args to be implemented.
 		unset( $this->args['customer_note'], $this->args['name'] );
 
+		$require_limit_data = ( $this->arg_isset( 'limit' ) || $this->arg_isset( 'page' ) || $this->arg_isset( 'offset' ) );
+
 		$this->build_query();
-		$this->run_query();
+		if ( ! $this->maybe_override_query() ) {
+			$this->run_query();
+		}
+	}
+
+	/**
+	 * Lets the `woocommerce_hpos_pre_query` filter override the query.
+	 *
+	 * @return boolean Whether the query was overridden or not.
+	 */
+	private function maybe_override_query(): bool {
+		/**
+		 * Filters the orders array before the query takes place.
+		 *
+		 * Return a non-null value to bypass the HPOS default order queries.
+		 *
+		 * If the query includes limits via the `limit`, `page`, or `offset` arguments, we
+		 * require the `found_orders` and `max_num_pages` properties to also be set.
```

Comment:
I don't think it's `required` anymore, but only encouraged.
```suggestion
		 * encourage the `found_orders` and `max_num_pages` properties to also be set.
```